korangar/src/graphics/passes/forward/rectangle.rs

```diff
@@ -0,0 +1,216 @@
+            push_constant_ranges: &[],
+        });
+
+        let pipeline = device.create_render_pipeline(&RenderPipelineDescriptor {
+            label: Some(DRAWER_NAME),
+            layout: Some(&pipeline_layout),
+            vertex: VertexState {
+                module: &shader_module,
+                entry_point: Some("vs_main"),
+                compilation_options: PipelineCompilationOptions::default(),
+                buffers: &[SimpleVertex::buffer_layout()],
+            },
+            fragment: Some(FragmentState {
+                module: &shader_module,
+                entry_point: Some("fs_main"),
+                compilation_options: PipelineCompilationOptions::default(),
+                targets: &[Some(ColorTargetState {
+                    format: render_pass_context.color_attachment_formats()[0],
+                    blend: None,
+                    write_mask: ColorWrites::default(),
+                })],
+            }),
+            multiview: None,
+            primitive: PrimitiveState {
+                topology: PrimitiveTopology::LineList,
+                ..Default::default()
+            },
+            depth_stencil: Some(DepthStencilState {
+                format: render_pass_context.depth_attachment_output_format()[0],
+                depth_write_enabled: false,
+                depth_compare: CompareFunction::Always,
+                stencil: StencilState::default(),
+                bias: DepthBiasState::default(),
+            }),
+            multisample: MultisampleState {
+                count: global_context.msaa.sample_count(),
+                ..Default::default()
+            },
+            cache: None,
+        });
+
+        Self {
+            vertex_buffer,
+            index_buffer,
+            instance_data_buffer,
+            bind_group_layout,
+            bind_group,
+            pipeline,
+            draw_count: 0,
+            instance_data: Vec::default(),
+        }
+    }
+
+    fn draw(&mut self, pass: &mut RenderPass<'_>, _draw_data: Self::DrawData<'_>) {
+        if self.draw_count == 0 {
+            return;
+        }
+
+        pass.set_pipeline(&self.pipeline);
+        pass.set_bind_group(2, &self.bind_group, &[]);
+        pass.set_vertex_buffer(0, self.vertex_buffer.slice(..));
+        pass.set_index_buffer(self.index_buffer.slice(..), IndexFormat::Uint16);
+        pass.draw_indexed(0..INDEX_COUNT as u32, 0, 0..self.draw_count as u32);
+    }
+}
+
+impl Prepare for ForwardRectangleDrawer {
+    fn prepare(&mut self, _device: &Device, instructions: &RenderInstruction) {
+        self.draw_count = instructions.rectangles.len();
+
+        if self.draw_count == 0 {
+            return;
+        }
+
+        self.instance_data.clear();
+
+        for instruction in instructions.rectangles.iter() {
+            self.instance_data.push(InstanceData {
+                world: instruction.world.into(),
+                color: instruction.color.into(),

```

Comment:
```suggestion
                color: instruction.color.components_linear(),
```

This was already wrong in the Circle/AABB code. So it should be updated there too. The "Color" values are curently in sRGB and they need to be converted the pre-multiplied alpha, linear colors.

Reply:
Does I make the change in Circle/AABB code also?

Reply:
That would be great!